src/plugins/ui_actions/public/cell_actions/components/cell_actions.tsx

```diff
@@ -0,0 +1,138 @@
+   */
+  name: string;
+  /**
+   * Field type.
+   * Example: 'keyword'
+   */
+  type: string;
+  /**
+   * Field value.
+   * Example: 'My-Laptop'
+   */
+  value: string;
+}
+
+export interface CellActionExecutionContext extends ActionExecutionContext {
+  /**
+   * Ref to a DOM node where the action can add custom HTML.
+   */
+  extraContentNodeRef: React.MutableRefObject<HTMLDivElement | null>;
+
+  /**
+   * Ref to the node where the cell action are rendered.
+   */
+  nodeRef: React.MutableRefObject<HTMLDivElement | null>;
+
+  /**
+   * Extra configurations for actions.
+   */
+  metadata?: Record<string, unknown>;
+
+  field: CellActionField;
+}
+
+export enum CellActionsMode {
+  HOVER_POPOVER = 'hover-popover',
+  ALWAYS_VISIBLE = 'always-visible',
+}
+
+export interface CellActionsProps {
+  /**
+   * Common set of properties used by most actions.
+   */
+  field: CellActionField;
+  /**
+   * The trigger in which the actions are registered.
+   */
+  triggerId: string;
+  /**
+   * UI configuration. Possible options are `HOVER_POPOVER` and `ALWAYS_VISIBLE`.
+   *
+   * `HOVER_POPOVER` shows the actions when the children component is hovered.
+   *
+   * `ALWAYS_VISIBLE` always shows the actions.
+   */
+  mode: CellActionsMode;
+  showTooltip?: boolean;
+  /**
+   * It shows 'more actions' button when the number of actions is bigger than this parameter.
+   */
+  showMoreActionsFrom?: number;
+  /**
+   * Custom set of properties used by some actions.
+   * An action might require a specific set of metadata properties to render.
+   * This data is sent directly to actions.
+   */
+  metadata?: Record<string, unknown>;
+}
+
+export const CellActions: React.FC<CellActionsProps> = ({
+  field,
+  triggerId,
+  children,
+  mode,
+  showTooltip = true,
+  showMoreActionsFrom = 3,
+  metadata,
+}) => {
+  const extraContentNodeRef = useRef<HTMLDivElement | null>(null);
+  const nodeRef = useRef<HTMLDivElement | null>(null);
+
+  const actionContext: CellActionExecutionContext = useMemo(
+    () => ({
+      field,
+      trigger: { id: triggerId },
+      extraContentNodeRef,
+      nodeRef,
+      metadata,
+    }),
+    [field, triggerId, metadata]
+  );
+
+  if (mode === CellActionsMode.HOVER_POPOVER) {
+    return (
+      <div ref={nodeRef} data-test-subj={'cellActions'}>
+        <HoverActionsPopover
+          actionContext={actionContext}
+          showTooltip={showTooltip}
+          showMoreActionsFrom={showMoreActionsFrom}
+        >
+          {children}
+        </HoverActionsPopover>
+
+        <div ref={extraContentNodeRef} />
+      </div>
+    );
+  }
+
+  return (
+    <div ref={nodeRef} data-test-subj={'cellActions'}>
+      {children}
+      <InlineActions
+        actionContext={actionContext}
+        showTooltip={showTooltip}
+        showMoreActionsFrom={showMoreActionsFrom}
+      />
+      <div ref={extraContentNodeRef} />
+    </div>
+  );
+};
+
+CellActions.displayName = 'CellActions';
```

Comment:
curious: How are you using this? Somehow for debugging production?

Reply:
`displayName` is for making sure the component name is present in the component inspector during development. It is deleted on production builds. But in this case, it is unnecessary because the component name is inferred when using `export const`. I will delete it.